src/arena.py

```diff
@@ -1 +1,108 @@
-WORKS = True
+import json
+import re
+import random
+from typing import List
+
+
+class Damage:
+    __x: int
+    __y: int
+    __z: int
+
+    def __init__(self, damage: str):
+        # XdY[ + Z]
+        m = re.search(r"^(\d+)d(\d+)( \+ (\d+))?$", damage)
+        if m:
+            self.__x = int(m.group(1))
+            self.__y = int(m.group(2))
+            z = m.group(4)
+            if z:
+                self.__z = int(z)
+            else:
+                self.__z = 0
+
+    def draw(self):
+        result = 0
+        for i in range(self.__x):
+            result += random.randint(1, self.__y)
+        result += self.__z
+        return result
+
+
+class Combatant:
+
+    name: str
+    last_damage: int
+    last_health: int
+    health: int
+    __damage: Damage
+
+    def __init__(self, name: str, hp: int, damage: Damage):
+        self.name = name
+        self.health = hp
+        self.__damage = damage
+
+    def attack(self) -> int:
+        self.last_damage = self.__damage.draw();
+        return self.last_damage
+
+    def get_attacked(self, damage: int) -> int:
+        self.last_health = self.health
+        self.health -= damage
+        if self.health < 0:
+            self.health = 0
+        return self.health
+
+    def is_dead(self):
```

Comment:
You could use a [property](https://realpython.com/python-descriptors/#python-descriptors-in-properties) here.

Reply:
ok